spec/models/reservation_spec.rb

```diff
@@ -10,8 +10,20 @@
   end
 
   it '#Reservation not valid with missing attributes' do
-    reservation = Reservation.new(room_id: 1, guest_id: 1, checkin_at: '2020-01-01')
+    reservation = create(:reservation)
+    reservation.checkin_at = nil
 
     expect(reservation).not_to be_valid
   end
+
+  it '#Reservation not valid with duplicate book code' do
```

Comment:
I like more validation messages such as `validates booking code uniqueness`. Tests verify that a class/method does something, so phrasing tests this way is usually easier to read. You could also wrap `it` blocks belonging to the same method in a `describe` block.

Reply:
Wrap it into a describe block will makes easier to organize, specially because there are more contexts to be tested soon after the next steps of implementation. 🚀